server/server.js

```diff
@@ -0,0 +1,44 @@
+const express = require('express');
+const bodyParser = require('body-parser');
+const app = express();
+const port = process.env.PORT || 5000;
+
+const events = [
+
+];
+
+let idCounter = 0;
+
+app.use(bodyParser.json());
+
+app.post('/events', function test2(req,res) {
+  idCounter += 1;
+  const eventObject = {
+    id: idCounter,
+    name: req.body.name
+  };
+  events.push(eventObject);
+  console.log(events);
+  res.status(201).send({ eventObject });
+});
+
+app.delete('/events/:id',(req, res) => {
+  const requestId = events.find((event) => event.id === req.params.id);
+  res.status(202).send(requestId);
+    return;
+}); 
+
+// create a GET route
+app.get('/events/:id', (req, res) => {
+    const event = events.find((event) => event.id === req.params.id);
```

Comment:
Challenge Question:
Imagine if this events list had a million entries. It would be awfully slow to traverse, which would cause an appearance of lag because of the slow response time. Do you know how we can make this call have a consistent response time, regardless of the number of events in the list?

PS: It's not worth spending too much time on this. The end goal is to put these events in a database, and not in memory, so whatever solution we have here will be deleted.

We'll go over it the next time we meet.

Hint: Look up big O notation